curator-framework/src/main/java/org/apache/curator/framework/state/ConnectionStateManager.java

```diff
@@ -240,27 +248,37 @@ private void processEvents()
     {
         try
         {
-            while ( !Thread.currentThread().isInterrupted() )
+            while ( !Thread.currentThread().isInterrupted() && (state.get() == State.STARTED) )
             {
-                final ConnectionState newState = eventQueue.take();
-
-                if ( listeners.size() == 0 )
+                int pollMaxMs = (sessionTimeoutMs * 2) / 3; // 2/3 of session timeout
```

Comment:
This is the configured session timeout MS. Will this cause issues if the negotiated timeout is different?

I guess it just means that it will potentially take 4/3 * configured session timeout to determine if the session has timed out during network partition?